Work around this GCC ICE: <https://bugs.gnu.org/31708>.  It shows up
only when doing native compiles on armhf-linux.

--- mariadb-10.1.33/plugin/semisync/semisync_master.cc	2018-07-28 02:13:12.604020250 +0200
+++ mariadb-10.1.33/plugin/semisync/semisync_master.cc	2018-07-28 02:14:11.907753417 +0200
@@ -847,6 +847,8 @@
   return function_exit(kWho, 0);
 }
 
+volatile const void *kSyncHeaderPtr = &ReplSemiSyncMaster::kSyncHeader;
+
 int ReplSemiSyncMaster::reserveSyncHeader(unsigned char *header,
 					  unsigned long size)
 {
@@ -873,7 +875,7 @@
     /* Set the magic number and the sync status.  By default, no sync
      * is required.
      */
-    memcpy(header, kSyncHeader, sizeof(kSyncHeader));
+    memcpy(header, (void *)kSyncHeaderPtr, sizeof(kSyncHeader));
     hlen= sizeof(kSyncHeader);
   }
   return function_exit(kWho, hlen);

